fluidkeys/main.go

```diff
@@ -267,6 +270,63 @@ func keyCreate() exitCode {
 	return 0
 }
 
+func keyList() exitCode {
+	fluidkeysDirectory, err := getFluidkeysDirectory()
+	if err != nil {
+		fmt.Printf("Failed to get fluidkeys directory")
+	}
+	db := database.New(fluidkeysDirectory)
+
+	keys, err := loadPgpKeys(db)
+	if err != nil {
+		panic(err)
+	}
+
+	emailAddressColumnWidth := len("Email address")
+
+	for _, key := range keys {
+		for _, id := range key.Identities {
+			emailAddressColumnWidth = Max(emailAddressColumnWidth, len(id.Name))
+		}
+	}
+
+	createdColumnWidth := 11
+	nextRotationColumnWidth := 13
+
+	header := fmt.Sprintf("%-*s  ", emailAddressColumnWidth, "Email address")
+	header += fmt.Sprintf("%-*s  ", createdColumnWidth, "Created")
+	header += fmt.Sprintf("%-*s  ", nextRotationColumnWidth, "Next rotation")
+
+	fmt.Printf("%s\n", colour.LightBlue(header))
+
+	printHorizontalRule(emailAddressColumnWidth, createdColumnWidth, nextRotationColumnWidth)
+
+	for _, key := range keys {
+		firstRow := true
+		for id := range key.Identities {
```

Comment:
Rather than looking up `firstRow` in every iteration of the loop (when you know it'll occur exactly once, at the start), I'd suggest dropping `firstRow` and doing this:
```
for _, key := range keys {
    firstUid = key.Identities[0]
    fmt.Printf("%-*s  ", emailAddressColumnWidth, firstUid)
    fmt.Printf("%-*s  ", createdColumnWidth, key.PrimaryKey.CreationTime.Format("2 Jan 2006"))
    fmt.Printf("\n")
    for uid := range key.Identities[1:] {
        fmt.Printf("%s\n", uid)
    }
}
```

Reply:
That would be *much* clearer, however, `Identities` is a map, not a list :/

Reply:
Ah, this is because `.key.Identities` is a map (which also means that uids are displayed in a random order, curiously). Suggest we convert into an (ordered) slice. Function coming up...

Reply:
Once #41 is merged you can do:
```go
for _, key := range keys {
    emails := key.Emails()
    if !len(emails) {
        continue
    }
    firstEmail = emails[0]
    fmt.Printf("%-*s  ", emailAddressColumnWidth, firstEmail)
    fmt.Printf("%-*s  ", createdColumnWidth, key.PrimaryKey.CreationTime.Format("2 Jan 2006"))
    fmt.Printf("\n")
    for email := range emails[1:] {
        fmt.Printf("%s\n", email)
    }
}
```